### PR TITLE
Make application binary smaller

### LINKED ISSUE #1005: [bounty] make app smaller
/bounty 50

atm there are tons of dependencies in the app we dont use

also

https://v2.tauri.app/concept/size/

### PR DESCRIPTION
**Pull request by Neptune650**
/claim #1005
/closes #1005

@louis030195 Makes the binary smaller by removing unused dependencies and making a profile.

#1008 doesn't work because it gets rid of an used library, and doesn't use the profile settings.

### CODE REVIEW

**Comment by louis030195** [comment]
@Neptune650 good

i was more thinking of dependencies in the package.json js stuff

also in the past i had experience product breaking with these rust stripping thing even though it compile (would remove too many stuff) but should be fine, did you check release-app.yml there are already some of these i think

also preferably it does not make the build long, i am constantly trying to reduce build time, even if it make app bigger for now

**Comment by Neptune650** [comment]
> @Neptune650 good
> 
> i was more thinking of dependencies in the package.json js stuff
> 
> also in the past i had experience product breaking with these rust stripping thing even though it compile (would remove too many stuff) but should be fine, did you check release-app.yml there are already some of these i think
> 
> also preferably it does not make the build long, i am constantly trying to reduce build time, even if it make app bigger for now 

@louis030195 I picked options that wouldn't lengthen up the build, after strip everything kept working for me so I think it's fine

I'll push a commit for getting rid of some package.json deps

File: screenpipe-js/package.json [hunk 1]
```diff
@@ -29,10 +29,8 @@
         "tsup": "^8.0.0"
     },
     "dependencies": {
-        "@scarf/scarf": "^1.4.0",
```
Comment (louis030195) [inline screenpipe-js/package.json:32]: this is used

does not seem so but it is, any way to mark it as used?

Comment (louis030195) [reply screenpipe-js/package.json:32]: its telemtry

**Comment by Neptune650** [comment]
@louis030195 Fixed, could you review.

**Pull request merged**